tests/rptest/tests/cloud_topics_test.py

```diff
@@ -42,8 +42,7 @@ def __create_initial_topics(self):
                              config={"redpanda.cloud_topic.enabled": "true"})
 
     @cluster(num_nodes=3)
-    @matrix(cloud_storage_type=[CloudStorageType.S3])
```

Comment:
Been a while for me digging deep into ducktape -- what's wrong with this? I thought this was fairly commonplace, or has this changed?

Actually looking around, should this be cloud_storage_type=get_cloud_storage_type(), or is leaving it off equivalent?

Reply:
>@matrix(cloud_storage_type=[CloudStorageType.S3])

my understanding is that this is that the test is giving an opinion about which provider to use, where as in CDT it is chosen by the infrastructure. specifically this test passes locally and in CI, but in CDT it fails trying to connect to minio, which doesn't exist.

>cloud_storage_type=get_cloud_storage_type()

get_cloud_storage_type is a free function that does not appear to affect the configuration of the test at all. so i think (1) cloud storage is setup automatically by the redpanda services with or without this, and (2) it is only useful if the test itself needs to reason about the cloud provider being used.

Reply:
> [INFO  - 2024-10-10 00:45:59,308 - redpanda - _load_s3_context - lineno:734]: No AWS credentials supplied, assuming minio defaults

Reply:
ok i think i fixed this. yeh having the storage type is good, but i had it pinned to s3 because i was working around a problem in c++ where i had set myself up to only work with s3.